### PR TITLE
Add worker tests

### LINKED ISSUE #32: Add workers.spec.js tests
Based on this tag https://github.com/microsoft/playwright/tree/v0.10.0. We don't need to get the tests green. We only need to incorporate the code. Let's also check if we can get code from Puppeteer

### PR DESCRIPTION
**Pull request by kblok**
closes #32

### CODE REVIEW

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 1]
```diff
@@ -152,6 +152,21 @@ public interface IPage
         /// </summary>
         event EventHandler<PageErrorEventArgs> PageError;
 
+        /// <summary>
+        /// Emitted when a dedicated WebWorker (<see href="https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API"/>) is spawned by the page.
+        /// </summary>
+        public event EventHandler<WorkerEventArgs> WorkerCreated;
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        event EventHandler<WorkerEventArgs> WorkerCreated;
```

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 2]
```diff
@@ -152,6 +152,21 @@ public interface IPage
...
 
+        /// <summary>
+        /// Emitted when a dedicated WebWorker (<see href="https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API"/>) is spawned by the page.
+        /// </summary>
+        public event EventHandler<WorkerEventArgs> WorkerCreated;
+
+        /// <summary>
+        /// Emitted when a dedicated WebWorker (<see href="https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API"/>) is terminated.
+        /// </summary>
+        public event EventHandler<WorkerEventArgs> WorkerDestroyed;
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        event EventHandler<WorkerEventArgs> WorkerDestroyed;
```

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 3]
```diff
@@ -152,6 +152,21 @@ public interface IPage
...
+
+        /// <summary>
+        /// Emitted when a dedicated WebWorker (<see href="https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API"/>) is terminated.
+        /// </summary>
+        public event EventHandler<WebsocketEventArgs> Websocket;
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        event EventHandler<WebsocketEventArgs> Websocket;
```

File: src/PlaywrightSharp.Abstractions/IPage.cs [hunk 4]
```diff
@@ -212,6 +227,11 @@ public interface IPage
         /// </summary>
         int DefaultNavigationTimeout { get; set; }
 
+        /// <summary>
+        /// Gets all workers in the page.
+        /// </summary>
+        public IWorker[] Workers { get; }
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IPage.cs]:
```suggestion
        IWorker[] Workers { get; }
```

File: src/PlaywrightSharp.Abstractions/IWorker.cs [hunk 5]
```diff
@@ -1,9 +1,39 @@
...
 {
     /// <summary>
     /// The Worker class represents a <see href="https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API">WebWorker</see>.
     /// </summary>
     public interface IWorker
     {
+        /// <summary>
+        /// Gets the Worker URL.
+        /// </summary>
+        public string Url { get; }
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IWorker.cs]:
```suggestion
        string Url { get; }
```

File: src/PlaywrightSharp.Abstractions/IPage.cs (hunk 1 above)
Comment (kblok) [reply src/PlaywrightSharp.Abstractions/IPage.cs]: We need an analyzer for this 😂

**Pull request merged**